pandas/core/sparse/series.py

```diff
@@ -146,10 +146,10 @@ def __init__(self, data=None, index=None, sparse_index=None, kind='block',
                 data = data._data
 
             elif isinstance(data, (Series, dict)):
+                data = Series(data, index=index)
                 if index is None:
```

Comment:
hmm, I think you might be able to remove this ``if index is None`` bizeness (and simply use data.index if needed below)